text/0000-overlapping_match.md

```diff
@@ -45,23 +45,24 @@ match val in {
 ```
 
 Benefits of this syntax:
-1. No new keywords need to be used. This is good thing since it means for a relatively small
-addition there would be no code breaks of existing code with this change.
-2. The `in` seems to imply that it sort of like an "iterator" of statements and will go through
-each of them in turn.
+1. Even though a new keyword has been made it will not break any code because Rust is a context
+sensitive language. And adding such a keyword increases the perceptual area of the new syntax
+so as to make it clear which type of match is being used.
+2. The word `fallthrough` is used because it implies that after a branch is finished then the
+control falls through to the check of the next branch.
 
 Meaning of parts:
 1. The `else` is used in a similar sort of vein to that of the `_` pattern in normal matches.
 The expression enclosed within this is only executed if none of the patterns within the
-`match/in` expression are matched. If `else` and `_` are both present then the code within the
+`match/fallthrough` expression are matched. If `else` and `_` are both present then the code within the
 `else` would be marked as unreadable.
 
 Edge cases:
 1. If the `_` pattern in present in any of the contained matches and the `else` block is also
 present then a `unreachable_code` lint is emitted on the code within the `else` block
 2. Since the main reason for using a `match` is the exhaustiveness checks as long as there isn't
 an `else` block then the compiler will output an error for `non-exhaustive patterns` if not all
-branches of the `match/in` are exhaustive.
+branches of the `match/fallthrough` are exhaustive.
 
```

Comment:
Another possibility would be to use `!` (never type) pattern instead of `else` to make the syntax more terse.
This would work because `let value: T = panic!();` where `T` is any type is valid - i.e: bottom inhabits every type.

Reply:
So would it be the following (for clarification):

```rust
match many x {
    _ if x % 5 == 0 => expr,
    ! => expr
}
```

Reply:
Yep.

Reply:
I definitely see the appeal of it, since as you say it is much more terse, but I think that it would be more confusing since it is in the area of pat/expr of the expression but does not follow the same "rules" as the other patterns since it will only be executed if no other patterns are executed sort of how the `else` block of an `if` statement is only executed if none of the other `if`s or `else if`s are executed